apps/ui/components/application/ApplicationPage.tsx

```diff
@@ -91,35 +97,64 @@ export function ApplicationPageWrapper({
   translationKeyPrefix,
   headContent,
   overrideText,
-  isDirty,
   children,
 }: ApplicationPageProps): JSX.Element {
-  const { t } = useTranslation();
+  const { t, i18n } = useTranslation();
   const router = useRouter();
+  const { asPath, push } = router;
 
   const pages = ["page1", "page2", "page3", "preview"] as const;
 
   const hideStepper =
     pages.filter((x) => router.asPath.match(`/${x}`)).length === 0;
-  const steps: StepperProps = {
-    steps: pages.map((x, i) => ({ slug: x, step: i })),
-    completedStep: application ? calculateCompletedStep(application) : 0,
-    basePath: getApplicationPath(application?.pk),
-    isFormDirty: isDirty ?? false,
+
+  const steps = pages.map((x, i) => ({
+    label: t(`application:navigation.${x}`),
+    state:
+      calculateCompletedStep(application) === i
+        ? StepState.available
+        : calculateCompletedStep(application) < i
+          ? StepState.disabled
+          : StepState.completed,
+  }));
+
+  const handleStepClick = (e: React.MouseEvent<HTMLButtonElement>) => {
+    const target = e.currentTarget;
+    const s =
+      Number(
+        target.getAttribute("data-testid")?.replace("hds-stepper-step-", "")
+      ) + 1;
+    if (s === 4 ? asPath.endsWith("preview") : asPath.includes(`page${s}`)) {
+      return; // already on the page, so do nothing
+    }
+    if (s === 4) {
+      push(`${getApplicationPath(application?.pk)}preview`);
+    } else {
+      push(`${getApplicationPath(application?.pk)}page${s}`);
+    }
   };
 
   const title = t(`${translationKeyPrefix}.heading`);
   const subTitle =
     headContent || overrideText || t(`${translationKeyPrefix}.text`);
+  const selectedStep = asPath.charAt(asPath.length - 1);
```

Comment:
Complete word comparisons if we have to convert strings to numbers.
If somebody changes the slug name this will break, without nothing indicating that you should have changed the comparison also.